### PR TITLE
Added ability to pass Edge instance to reachable method

### PR DESCRIPTION
**Pull request by mrunderline**
@Thenerdstation, it's almost done! feel free to comment or ask any changes.

### CODE REVIEW

File: tensornetwork/network_operations.py [hunk 1]
```diff
@@ -590,21 +590,23 @@ def _reachable(nodes: Set[BaseNode]) -> Set[BaseNode]:
...
 
 
-def reachable(nodes: Union[BaseNode, Iterable[BaseNode]]) -> Set[BaseNode]:
+def reachable(inputs: Union[BaseNode, Iterable[BaseNode], Edge, Iterable[Edge]]) -> Set[BaseNode]:
   """
-  Computes all nodes reachable from `node` by connected edges.
+  Computes all nodes reachable from `node` or `edge.node1` by connected edges.
   Args:
-    nodes: A `BaseNode` or collection of `BaseNodes`
+    inputs: A `BaseNode`/`Edge` or collection of `BaseNodes`/`BaseEdges`
```
Comment (Thenerdstation) [inline tensornetwork/network_operations.py]: No such thing as BaseEdges

Comment (mrunderline) [reply tensornetwork/network_operations.py]: Ops! it's been changed and corrected!

File: tensornetwork/network_operations.py [hunk 2]
```diff
@@ -590,21 +590,23 @@ def _reachable(nodes: Set[BaseNode]) -> Set[BaseNode]:
...
   Raises:
     ValueError: If an unknown value for `strategy` is passed.
   """
 
-  if isinstance(nodes, BaseNode):
-    nodes = {nodes}
-  return _reachable(set(nodes))
+  if isinstance(inputs, BaseNode):
+    inputs = {inputs}
+  elif isinstance(inputs, Edge):
```
Comment (Thenerdstation) [inline tensornetwork/network_operations.py]: Do you want to also support the case where it is an iterable of `Edge`s?

**Comment by mrunderline** [comment]
I think it's almost done.
A very very simple test added; cause i'm still beginner at it 😅

File: tensornetwork/tests/network_operations_test.py [hunk 3]
```diff
@@ -218,6 +218,13 @@ def test_reachable(backend):
   _ = [nodes[n][0] ^ nodes[n + 1][1] for n in range(len(nodes) - 1)]
   assert set(nodes) == tn.reachable(nodes[0])
 
+  edges = []
```
Comment (Thenerdstation) [inline tensornetwork/tests/network_operations_test.py]: Dont append this to an existing test, Instead, create a new test.

Also, I would highly recommend not creating an `Edge` object directly like this, as the `Node`s that the edge is trying connect won't exist in those `Node`s. Instead, you should create edges via the `tn.connect` operation. (You can also use the `^` symbol if you don't like typing.) See the other tests or our tutorial on `tensornetwork.readthedocs.io` for examples.

Comment (mrunderline) [reply tensornetwork/tests/network_operations_test.py]: thanks for your attention and kind comments.
a new commit pushed, please check it out.

File: tensornetwork/network_operations.py [hunk 4]
```diff
@@ -590,21 +590,25 @@ def _reachable(nodes: Set[BaseNode]) -> Set[BaseNode]:
...
+    inputs = {inputs.node1}
+  elif isinstance(inputs, list) and all(isinstance(_, Edge) for _ in inputs):
```
Comment (Thenerdstation) [inline tensornetwork/network_operations.py]: Can you change the `_` to `x`? Generally `_` is only used when we want to ignore the variable, but in this case we are actually using it.

Comment (Thenerdstation) [reply tensornetwork/network_operations.py]: Same just below.

**Pull request merged**